sky/clouds/azure.py

```diff
@@ -323,7 +325,9 @@ def _failover_disk_tier() -> Optional[resources_utils.DiskTier]:
             'zones': None,
             **image_config,
             'disk_tier': Azure._get_disk_type(_failover_disk_tier()),
-            'cloud_init_setup_commands': cloud_init_setup_commands
+            'cloud_init_setup_commands': cloud_init_setup_commands,
+            'azure_subscription_id': self.get_project_id(dryrun),
+            'resource_group': f'{cluster_name_on_cloud}-{region_name}',
```

Comment:
Seems like we are using `cluster_name` originally instead of `cluster_name_on_cloud`. Is it intended? Will it cause any backward compatibility issues?

Reply:
Yes, this is intended, as we should have used `cluster_name_on_cloud` instead of `cluster_name`, because the former should be the string used on the cloud to avoid length limit issue. It will not cause backward compatibility issue, as we will always restore the old config for existing cluster in: https://github.com/skypilot-org/skypilot/blob/1be4d98abcc436066529d2612530990aa1c6f729/sky/backends/backend_utils.py#L681-L733